src/components/SignInButton/SignInButton.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEthereum } from '../../contexts/EthereumContext'
+import { useNetwork } from '@/contexts/NetworkContext'
+import { useProfile } from '../../contexts/ProfileContext'
+
+/**
+ * Provides a button for signing in with a wallet or Universal
+ * Profile and proofing ownership over an address. It leverages
+ * the useEthereum hook from the EthereumContext
+ * for managing blockchain connections.
+ *
+ * @returns {JSX.Element} - A JSX structure of the button
+ */
+function SignInButton() {
+  const { account, verified, signInWithEthereum } = useEthereum()
+  const { network } = useNetwork()
+  const { profile } = useProfile()
+
+  let buttonText
+  let buttonAction
+  let buttonClass
```

Comment:
Please add: https://prettier.io/docs/en/options.html#semicolons

in prettier or turn it to true so it matches our other repos.